### PR TITLE
Accordion: remove commons type

### PR DESCRIPTION
**Pull request by smhigley**
Parent issue: #22862

While removing Commons, I noticed a couple places there was a prop on state that wasn't used, so I removed those. I also alphabetized type props.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                   |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                   Change |
| :---------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `73.795 kB`<br />`22.514 kB` | `73.722 kB`<br />`22.493 kB` | `-73 B` <br />`-21 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/54a23bb67bc23d6e45017a4ac6039f0160ce7ec8'>54a23bb67bc23d6e45017a4ac6039f0160ce7ec8</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/Avatar-mount_base.html">1221</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/Avatar-mount.html">1147</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/Button-mount_base.html">719</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/Button-mount.html">708</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProvider-mount_base.html">2104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProvider-mount.html">2130</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-mount_base.html">374</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-mount.html">380</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">295</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">360</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/MakeStyles-mount_base.html">1830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22863/perf-test-react-components/MakeStyles-mount.html">1880</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-accordion/src/components/AccordionItem/AccordionItem.types.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import * as React from 'react';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
-export type AccordionItemContextValue = Omit<AccordionItemCommons, 'value'> & {
-  open: boolean;
+export type AccordionItemContextValue = {
+  disabled: boolean;
```
Comment (khmakoto) [inline packages/react-accordion/src/components/AccordionItem/AccordionItem.types.ts:5]: This should probably still be in terms of `AccordionItemProps`:

```suggestion
export type AccordionItemContextValue = Required<Pick<AccordionItemProps, disabled>> & {
```

Comment (smhigley) [reply packages/react-accordion/src/components/AccordionItem/AccordionItem.types.ts:5]: haha I actually wrote that first, stared at it for about 10s, and then removed it :D.

I was primarily going back and forth on whether the ContextValue was conceptually tied to props, or its own thing. I'm fine switching it back though :)

**Review by khmakoto — APPROVED** [review]

File: packages/react-accordion/src/components/Accordion/useAccordion.ts [hunk 2]
```diff
@@ -76,12 +70,14 @@ function initializeUncontrolledOpenItems({
...
- * @param param2 - {multiple, collapsible}
+ * @param multiple - if Accordion support multiple Panels opened at the same time
+ * @param collapsible - if Accordion support multiple Panels closed at the same time
  */
 function updateOpenItems(
   value: AccordionItemValue,
   previousOpenItems: AccordionItemValue[],
-  { multiple, collapsible }: Pick<AccordionState, 'multiple' | 'collapsible'>,
+  multiple: boolean,
+  collapsible: boolean,
```
Comment (ling1726) [inline packages/react-accordion/src/components/Accordion/useAccordion.ts:80]: Why was this change necessary ?

Comment (smhigley) [reply packages/react-accordion/src/components/Accordion/useAccordion.ts:80]: I can revert it if you'd like! I was wondering if `multiple` and `collapsible` were only passed in as an object so they could be typed as `AccordionCommons`. I could also do a `Required<Pick<AccordionProps, 'multiple' | 'collapsible'>>`, but that seemed a little over-engineered for two boolean props. I could also type them as `{ multiple: boolean; collapsible: boolean; }` if there was another reason to pass them in as an object.

@ling1726 do you have a preference on the approach?

Comment (ling1726) [reply packages/react-accordion/src/components/Accordion/useAccordion.ts:80]: ok I see, you wanted to avoid using the TS helpers here. I'm ok with that

**Review by ling1726 — APPROVED** [review]
